common/src/main/java/dev/aurelium/auraskills/common/source/SourceLoader.java

```diff
@@ -201,8 +201,18 @@ public Map<SourceTag, List<XpSource>> loadTags(Skill skill, File contentDirector
         }
     }
 
-    private Collection<XpSource> getFilteredSource(List<XpSource> sources, String filter) {
-        return sources.stream().filter(source -> source.name().toLowerCase().contains(filter.replace("!", "").replace("*", "").toLowerCase())).collect(Collectors.toList());
+    private Collection<XpSource> getFilteredSource(List<XpSource> sources, String rawFilter) {
+        String filter = rawFilter.replace("!", "").replace("*", "").toLowerCase();
+        Boolean isEndingWildcard = rawFilter.endsWith("*");
+        Boolean isBeginningWildcard = rawFilter.startsWith("*") || rawFilter.startsWith("!");
```

Comment:
Use a primitive boolean here instead of the boxed type. This recommendation applies generally across the project unless you need to specifically represent a `null` state.